packages/@react-spectrum/dnd/docs/dragAndDrop.mdx

```diff
@@ -0,0 +1,246 @@
+and provide those hooks with your desired options.
+
+`useDragHooks` supports the following options:
+<TypeContext.Provider value={dndDocs.links}>
+  <InterfaceType properties={dndDocs.exports.DraggableCollectionProps.properties} />
+</TypeContext.Provider>
+
+`useDropHooks` supports the following options:
+<TypeContext.Provider value={dndDocs.links}>
+  <InterfaceType properties={dndDocs.exports.DroppableCollectionProps.properties} />
+</TypeContext.Provider>
+
+`useDragHooks` will then provide you with a set of drag hooks that you can pass to the component via its `dragHooks` prop, thus enabling drag operations for the component.
+Drop operations can be enabled in the same way by passing the hooks from `useDropHooks` to the component's `dropHooks` prop.
+
+You may be wondering why we didn't simply import the drag and drop hooks within the components themselves. The reason behind this is to reduce the bundle size of
+the component for users who don't need the drag and drop support for their component.
+
+The following list shows which components currently support drag and drop:
+-  [ListView](ListView.html#drag-and-drop)
+
+## Example
+
+The example below illustrates how one would use `useDragHooks` and `useDropHooks` to enable dragging files from one ListView to another. For the first ListView, we want to make the rows draggable and
+have the dragged rows removed from the list upon a successful drop. To accomplish this, we first want to specify the types of data associated with each item via the array returned by `getItems`. In this case,
+we add a `text/plain` type that matches the text value associated with each item. The item's type is also included with the item's `id` mapped to it. We also provide a `onDragEnd` event handler to `useDragHooks`
+that removes the dragged items from the list only upon a successful drop operation.
+
+<details>
+  <summary style={{fontWeight: 'bold'}}><ChevronRight size="S" /> Show code</summary>
+
+```tsx example export=true render=false
+import Folder from '@spectrum-icons/workflow/Folder';
+import {Item, ListView} from '@react-spectrum/list';
+import {Text} from '@adobe/react-spectrum';
+import {useDragHooks} from '@react-spectrum/dnd';
+import {useListData} from '@react-stately/data';
+
+function DraggableList(props) {
+  let {
+    items,
+    ...otherProps
+  } = props;
+  let list = useListData({
+    initialItems: items || [
+      {id: 'a', textValue: 'Adobe Photoshop', type: 'file'},
+      {id: 'b', textValue: 'Adobe XD', type: 'file'},
+      {id: 'c', textValue: 'Documents', type: 'folder'},
+      {id: 'd', textValue: 'Adobe InDesign', type: 'file'},
+      {id: 'e', textValue: 'Utilities', type: 'folder'}
+    ]
+  });
+
+  let getItems = (keys) => [...keys].map(key => {
+    let item = list.getItem(key);
+    return {
+      'text/plain': item.textValue,
+      [item.type]: item.id
+    };
+  });
```

Comment:
perhaps I'm misusing this here? Just trying to imagine the best way to communicate an item's non-text information since the stuff returned here shows up in the item's types in onDrop and then accessed via getText. Kinda feels weird passing the `id` like this

Perhaps I am misrepresenting the items of the ListView? Looking at https://github.com/adobe/react-spectrum/blob/b4977a1d6279337d75d6b8b5c2645ffbb88fdee8/packages/%40react-types/shared/src/dnd.d.ts#L56-L74

Reply:
JSON.stringify it